xdsl/ir.py

```diff
@@ -630,6 +630,29 @@ def __eq__(self, other: object) -> bool:
     def __hash__(self) -> int:
         return id(self)
 
+    def __str__(self) -> str:
+        from xdsl.printer import Printer
+        res = StringIO()
+        printer = Printer(stream=res)
+        printer.print_op(self)
+        desc = res.getvalue()
+        # Printer currently prints a new line when printing an operation.
+        # We don't want this behaviour when printing ops inline, so remove
+        # the trailing newline for now
+        last_char_is_newline = not desc or desc[-1] == "\n"
+        assert last_char_is_newline, "If the format changes, update this function"
+        desc = desc[:-1]
```

Comment:
I would very much appreciate if we could fix this in a later PR. Having this kind of code in the repo feels bad to me.

Reply:
#450